concordium-consensus/src/Concordium/GlobalState/Persistent/Instances.hs

```diff
@@ -449,27 +549,46 @@ updateContractInstance fupd addr (InstancesTree s it0) = upd baseSuccess (contra
                 in upd newCont (i - 2^h) =<< mproject r
             | otherwise = return Nothing
 
-allInstances :: forall m. MonadBlobStore m => Instances -> m [PersistentInstance]
+allInstances :: forall m pv. (IsProtocolVersion pv, MonadBlobStore m) => Instances pv -> m [PersistentInstance pv]
 allInstances InstancesEmpty = return []
 allInstances (InstancesTree _ it) = mapReduceIT mfun it
     where
         mfun (Left _) = return mempty
         mfun (Right inst) = return [inst]
 
-makePersistent :: forall m. MonadBlobStore m => Modules.Modules -> Transient.Instances -> m Instances
+makePersistent :: forall m pv. MonadBlobStore m => Modules.Modules -> Transient.Instances -> m (Instances pv)
 makePersistent _ (Transient.Instances Transient.Empty) = return InstancesEmpty
 makePersistent mods (Transient.Instances (Transient.Tree s t)) = InstancesTree s <$> conv t
     where
-        conv :: Transient.IT -> m (BufferedBlobbed BlobRef IT)
+        conv :: Transient.IT -> m (BufferedBlobbed BlobRef (IT pv))
         conv (Transient.Branch lvl fll vac hsh l r) = do
             l' <- conv l
             r' <- conv r
             makeBufferedBlobbed (Branch lvl fll vac hsh l' r')
         conv (Transient.Leaf i) = convInst i >>= makeBufferedBlobbed . Leaf
         conv (Transient.VacantLeaf si) = makeBufferedBlobbed (VacantLeaf si)
-        convInst Transient.Instance{instanceParameters=Transient.InstanceParameters{..}, ..} = do
+        convInst (Transient.InstanceV0 Transient.InstanceV {_instanceVParameters=Transient.InstanceParameters{..}, ..}) = do
+            pIParams <- makeBufferedRef $ PersistentInstanceParameters{
+                pinstanceAddress = _instanceAddress,
+                pinstanceOwner = instanceOwner,
+                pinstanceContractModule = GSWasm.miModuleRef instanceModuleInterface,
+                pinstanceInitName = instanceInitName,
+                pinstanceParameterHash = instanceParameterHash,
+                pinstanceReceiveFuns = instanceReceiveFuns
+            }
+            -- This pattern is irrefutable because if the instance exists in the Basic version,
```

Comment:
Surely I miss something here - but how come?

Reply:
Because that is an invariant in the basic state, and we converted modules from the basic state to the persistent one already.

Reply:
I added this comment to the code.